crates/rust/src/bindgen.rs

```diff
@@ -443,7 +443,7 @@ impl Bindgen for FunctionBindgen<'_, '_> {
                                     .as_deref()
                                     .unwrap()
                                     .to_upper_camel_case();
-                                format!("&*({op} as u32 as usize as *const {name})")
+                                format!("Option::as_ref(&*({op} as u32 as usize as *const Option<{name}>)).unwrap()")
```

Comment:
Mind adding a helper method to `Resource<T>` which is `#[doc(hidden)]` to call here? That way this could avoid duplicating knowledge of the internal representation across the code generator and the actual source